doc/decisions/database_structure.md

```diff
@@ -0,0 +1,78 @@
+# Database Structure
+
+The main goal of this document is to describe the database structure/layout especially in terms of relations between data entities.
+
+## Problem
+
+Semantically the plant information can be described as a hierarchy of entities. E.g., a plant is always a member on the variety level, while a variety is a part of species, species is a part of genus, genus is a part of subfamily, subfamily is a part of family. Please take a look at the diagram below. Each of these entities can have their own properties, e.g., a variety under a specie can have different height than the specie that it belongs to.
+
+The challenge is to define the structure of the database in a way that the data can be easily queried and updated.
+
+Another goal is to be able to define relations between entities, e.g., a plant of one variety doesn't want to be neighbor to plant of some other variety.
```

Comment:
Actually this is a Constraint for the decision, like all other features we want for the hierarchy.